MapboxCoreNavigation/RouteController.swift

```diff
@@ -392,22 +409,20 @@ open class RouteController: NSObject {
             averageRelativeAngle = (relativeAnglepointBehind + relativeAnglepointAhead) / 2
         }
         
-        let calculatedCourseForLocationOnStep = (wrappedCourse + averageRelativeAngle).wrap(min: 0, max: 360)
-        
-        var userCourse = calculatedCourseForLocationOnStep
-        var userCoordinate = closest.coordinate
-        
-        if location.course >= 0 && location.speed >= RouteControllerMinimumSpeedForLocationSnapping {
-            if calculatedCourseForLocationOnStep.differenceBetween(location.course) > RouteControllerMaxManipulatedCourseAngle && location.horizontalAccuracy < 20 {
-                userCourse = location.course
-                
-                if closest.distance > RouteControllerUserLocationSnappingDistance && location.horizontalAccuracy < 20 {
-                    userCoordinate =  location.coordinate
-                }
-            }
+        return (wrappedCourse + averageRelativeAngle).wrap(min: 0, max: 360)
+    }
+    
+    /**
+     Determines if the a location is qualified enough to allow the user puck to become unsnapped.
+     */
+    func shouldAllowUnsapped(location: CLLocation, given snappedCourse: CLLocationDirection) -> Bool {
```

Comment:
Typo: “unsnapped”. A more grammatical name would be `shouldSnap(_:facing:)`. “Given” isn’t a word that appears in Swift method names. Note that the logic would be inverted, since positive conditions are easier to reason about than negative conditions.